solr/core/src/java/org/apache/solr/filestore/DistribPackageStore.java

```diff
@@ -572,7 +578,16 @@ private void ensurePackageStoreDir(Path solrHome) {
   }
 
   public static Path getPackageStoreDirPath(Path solrHome) {
-    return solrHome.resolve(PackageStoreAPI.PACKAGESTORE_DIRECTORY);
+    var path = solrHome.resolve(PackageStoreAPI.PACKAGESTORE_DIRECTORY);
+    if (!Files.exists(path)) {
```

Comment:
I think we need a synchronized block to ensure that no race conditions happen

Reply:
Done